drv/ArrayList.drv

```diff
@@ -587,6 +601,59 @@ public class ARRAY_LIST KEY_GENERIC extends ABSTRACT_LIST KEY_GENERIC implements
 		return modified;
 	}
 
+	@Override
+	public boolean retainAll(final Collection<?> c) {
```

Comment:
If `c` is an instance of `COLLECTION` this should delegate to the type-specific call to avoid boxing

Reply:
Technically you would have to define it as collection<?> first to actually cause a case like that and have the boxing problem beforehand.
On top of that same question again: should this apply to removeAll too? Because it is just a copypaste job and removing 1 character from the function.

Reply:
> Technically you would have to define it as collection<?> first to actually cause a case like that and have the boxing problem beforehand.

What? As soon as this method is entered there is boxing.

Yeah I guess it makes sense to have it in removeAll, too.

Reply:
You have to define it a List<Integer> instead of a IntList to enforce  that Collection<Integer> is called instead of IntCollection.
How do you test in fastutil for a collection type?

Reply:
Well, yes, but if `IntList` is used as an implementation of `Collection<Integer>` for example, this method is called.

I guess `if (c instanceof COLLECTION)`?

Reply:
If it is defined as Collection yes, otherwise the most top interface applies and since IntCollection is above Collection that one gets used. You have to choose to do it is all i am saying.

Reply:
Huh? Above? The reference is resolved at compile time, the most specific method is used. IntCollection is "below" Collection in the sense that it is more specific.

You don't necessarily have to choose to do so when you as soon as you are working on either side of an API where the other side doesn't know about fastutil.

Anyway, the check is cheap for a bulk operation, so it won't hurt to have it there I guess.